### PR TITLE
fix(nuxt): always use increment for id with client side islands

### PR DESCRIPTION
**Pull request by bounoable**
### 🔗 Linked issue

Issue: #20988
PR: #20992

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

I still get the webCrypto.randomUUID() error when navigating to a page with a server component in dev mode. I think #20992 missed to replace a `randomUUID()` call because this fix seems to fix work.

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by danielroe — APPROVED** [review]

**Comment by danielroe** [comment]
Would you provide a reproduction of the bug this change solves? 🙏

File: packages/nuxt/src/app/components/nuxt-island.ts [hunk 1]
```diff
@@ -86,7 +86,7 @@ export default defineComponent({
       ssrHTML.value = renderedHTML
     }
     const slotProps = computed(() => getSlotProps(ssrHTML.value))
-    const uid = ref<string>(ssrHTML.value.match(SSR_UID_RE)?.[1] ?? randomUUID())
+    const uid = ref<string>(ssrHTML.value.match(SSR_UID_RE)?.[1] ?? getId())
```
Comment (huang-julien) [inline packages/nuxt/src/app/components/nuxt-island.ts:89]: Not quite sure, but I did have a hydration mismatch. i'll investigate

**Comment by bounoable** [comment]
I tried but this actually cannot reproduced on Stackblitz & co. because webCrypto is only non-available in non-https environments 🥲

This error is triggered within my application when "navigating to another page that needs to render a server component on the client". I'm using the `.server.vue` suffix, not the `<NuxtIsland>` component.

File: packages/nuxt/src/app/components/nuxt-island.ts (hunk 1 above)
Comment (huang-julien) [reply packages/nuxt/src/app/components/nuxt-island.ts:89]: I'm resolving this as i can't reproduce it anymore...

**Review by huang-julien — APPROVED** [review]
I think this should be OK. There shouldn't be any risk of having a conflict between ids since we still use `randomUUID` server side.
The issue here is that `randomUUID` isn't available in non-secure context. 
I can't reproduce it locally even with ngrok, and it seems like `localhost` is a secure context by default ?

What do you think @danielroe ?

BTW you can update snapshots with `pnpm test:fixtures --u` 🙂

**Pull request merged**